diff --git a/CMSSW.py b/CMSSW_new.py
index 2b25437..5ba106f 100644
--- a/CMSSW.py
+++ b/CMSSW_new.py
@@ -1,3 +1,11 @@
 import FWCore.ParameterSet.Config as cms
 import pickle
 process = pickle.load(open('CMSSW.py.pkl', 'rb'))
+
+# Keep track of the most important random number generator seeds
+from IOMC.RandomEngine.RandomServiceHelper import RandomNumberServiceHelper
+randSvc = RandomNumberServiceHelper(process.RandomNumberGeneratorService)
+print "generator seed =", randSvc.getNamedSeed("generator")
+print "g4SimHits seed =", randSvc.getNamedSeed("g4SimHits")
+print "VtxSmeared seed =", randSvc.getNamedSeed("VtxSmeared")
+print "mix seed =", randSvc.getNamedSeed("mix")
